tools/wpt/browser.py

```diff
@@ -431,8 +431,45 @@ class Servo(Browser):
     product = "servo"
     requirements = "requirements_servo.txt"
 
+    def platform_string(self):
+        platform = {
+            "Linux": "linux",
+            "Windows": "win",
+            "Darwin": "mac"
+        }.get(uname[0])
+
+        if platform is None:
+            raise ValueError("Unable to construct a valid Servo package for current platform")
+
+        return "%s" % (platform)
+
+    def find_extension(self, platform):
+        if platform == "linux":
+            extension = ".tar.gz"
+        elif platform == "win":
+            extension = ".msi"
+        else:
+            extension = ".dmg"
+
+        return "%s" % (extension)
+
     def install(self, dest=None):
-        raise NotImplementedError
+        """Install latest Browser Engine."""
+        if dest is None:
+            dest = os.pwd
+
+        extension = self.find_extension(self.platform_string())
+        url = "https://download.servo.org/nightly/%s/servo-latest%s" % (self.platform_string(), extension)
```

Comment:
Rather than two separate functions, if there is a single function that returns (platform, extension), we can do:
```
def platform_components()
    ...
    return (platform, extension)

url = "https://download.servo.org/nightly/%s/servo-latest%s" % self.platform_components()
```